File: java/src/test/java/org/sammancoaching/conditionals/GuardClausesTest.java

package org.sammancoaching.conditionals;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import static org.junit.jupiter.api.Assertions.assertEquals;

class GuardClausesTest {

    private GuardClauses guard;

    @BeforeEach
    void setUp() {
        guard = new GuardClauses();
    }

    @ParameterizedTest
    @CsvSource({
            "4, 3, 3",    // expected y, expected z, input
            "0, 0, 9"
    })
    void test_introduce_guard_clause_simple(int expectedY, int expectedZ, int input) {
        guard.y = 0;
        guard.z = 0;
        guard.introduce_guard_clause_simple(input);
        assertEquals(expectedY, guard.y);
        assertEquals(expectedZ, guard.z);
    }

    @ParameterizedTest
    @CsvSource({
            "4, 3, 3",    // expected y, expected z, input
            "0, 0, 9"
    })
    void test_remove_guard_clause_simple(int expectedY, int expectedZ, int input) {
        guard.y = 0;
        guard.z = 0;
        guard.remove_guard_clause_simple(input);
        assertEquals(expectedY, guard.y);
        assertEquals(expectedZ, guard.z);
    }

    @ParameterizedTest
    @CsvSource({
            "0, 3",
            "40, 9"
    })
    void test_remove_guard_clause_multi_return(int expected, int input) {
        guard.y = 0;
        int result = guard.remove_guard_clause_multi_return(input);
        assertEquals(expected, result);
    }

    @ParameterizedTest
    @CsvSource({
            "0, 3",
            "40, 9"
    })
    void test_introduce_guard_clause_multi_return(int expected, int input) {
        guard.y = 0;
        int result = guard.introduce_guard_clause_multi_return(input);
        assertEquals(expected, result);
    }
}

File: csharp/RefactorConditionals/RefactorConditionals.Tests/GuardClausesTest.cs
using Xunit;

namespace RefactorConditionals.Tests
{
    public class GuardClausesTest
    {
        private GuardClauses guard;

        public GuardClausesTest()
        {
            guard = new GuardClauses();
        }

        [Theory]
        [InlineData(4, 3, 3)]    // expected y, expected z, input
        [InlineData(0, 0, 9)]
        public void TestIntroduceGuardClauseSimple(int expectedY, int expectedZ, int input)
        {
            guard.y = 0;
            guard.z = 0;
            guard.IntroduceGuardClauseSimple(input);
            Assert.Equal(expectedY, guard.y);
            Assert.Equal(expectedZ, guard.z);
        }

        [Theory]
        [InlineData(4, 3, 3)]    // expected y, expected z, input
        [InlineData(0, 0, 9)]
        public void TestRemoveGuardClauseSimple(int expectedY, int expectedZ, int input)
        {
            guard.y = 0;
            guard.z = 0;
            guard.RemoveGuardClauseSimple(input);
            Assert.Equal(expectedY, guard.y);
            Assert.Equal(expectedZ, guard.z);
        }

        [Theory]
        [InlineData(0, 3)]
        [InlineData(40, 9)]
        public void TestRemoveGuardClauseMultiReturn(int expected, int input)
        {
            guard.y = 0;
            int result = guard.RemoveGuardClauseMultiReturn(input);
            Assert.Equal(expected, result);
        }

        [Theory]
        [InlineData(0, 3)]
        [InlineData(40, 9)]
        public void TestIntroduceGuardClauseMultiReturn(int expected, int input)
        {
            guard.y = 0;
            int result = guard.IntroduceGuardClauseMultiReturn(input);
            Assert.Equal(expected, result);
        }
    }
}